zebrad/tests/common/lightwalletd/send_transaction_test.rs

```diff
@@ -123,6 +123,40 @@ pub async fn run() -> Result<()> {
         assert_eq!(response, expected_response);
     }
 
+    // We test mempool grpc calls here as we know we have mempool transactions
+
+    // Transactions can be in the RPC queue, lets wait until they are retired
+    // before checking the mempool.
+    tokio::time::sleep(tokio::time::Duration::from_secs(120)).await;
+
+    // Call `GetMempoolStream`
+    let mut response = rpc_client
+        .get_mempool_stream(wallet_grpc::Empty {})
+        .await?
+        .into_inner();
+
+    // Make sure transactions are returned from the mmepool
+    let mut counter = 0;
+    while let Some(_txs) = response.message().await? {
+        counter += 1;
+    }
+    assert!(counter > 0);
```

Comment:
This assertion might be failing because lightwalletd is still syncing when the transactions are sent. When lightwalletd finishes syncing, it looks like it clears its internal mempool cache, so transactions are no longer available to clients.

(It should eventually re-check the mempool and get the transactions again. But the test will still be unreliable, because it doesn't wait.)

> {"app":"lightwalletd","level":"info","msg":"Latest Block changed, clearing everything","time":"2022-08-01T05:30:30Z"}

Reply:
I opened ticket #4894 to fix this.